src/montgomery.rs

```diff
@@ -193,6 +193,81 @@ pub(crate) fn elligator_encode(r_0: &FieldElement) -> MontgomeryPoint {
     MontgomeryPoint(u.to_bytes())
 }
 
+/// Perform the inverse Elligator2 mapping from a Montgomery point to
+/// field element. This algorithm is based on [Elligator:
```

Comment:
nit: the indefinite article
```suggestion
/// a field element. This algorithm is based on [Elligator:
```